app/main.py

```diff
@@ -1,3 +1,37 @@
+import json
+import os
+from app.customer import Customer
+from app.shop import Shop
+
+
 def shop_trip():
-    # write your code here
-    pass
+    config_path = os.path.join(os.path.dirname(__file__), "config.json")
+
+    with open(config_path) as f:
+        config = json.load(f)
+
+    fuel_price = config["FUEL_PRICE"]
+    customers = [Customer(**customer) for customer in config["customers"]]
+    shops = [Shop(**shop) for shop in config["shops"]]
```

Comment:
Ensure that the `config['shops']` contains dictionaries with keys matching the parameters expected by the `Shop` class constructor. Otherwise, this line will raise a TypeError.